packages/airnode-deployer/src/entrypoint.ts

```diff
@@ -0,0 +1,34 @@
+const cloudProvider = process.env.AIRNODE_CLOUD_PROVIDER;
+
+if (!cloudProvider) {
+  throw new Error('Missing cloud specification');
+}
+
+async function cloudHandler() {
+  return await import(/* webpackIgnore: true */ `./handlers/${cloudProvider}`);
```

Comment:
> Can you explain why we need the `webpackIgnore`?

It's used so that when the webpack is called for the `entrypoint.ts` (or rather `entrypoint.js`) it won't follow the dynamic imports and won't bundle everything into `.webpack/index.js`.

> Also I think `switch` case instead of dynamic import is simpler (and better) solution.

It was done this way for two reasons.
First, this way you won't include AWS-specific node code for GCP and vice versa. Which I guess would not be such a big deal.
Second, I wanted to keep the resulting directory structure (`handlers/aws/index.js`) pretty much the same. Some code expects this structure (when looking for `config.json` and validator templates for example) and I think it's currently easier to keep it than refactoring it. Unfortunately, GCP has a limitation that the main cloud function file must be at the root, that's why I added the `entrypoint.ts` and bundled it separately from actual cloud-specific handlers.

Reply:
It's worth adding a code comment to explain this for future readers imo

Reply:
> First, this way you won't include AWS-specific node code for GCP and vice versa. Which I guess would not be such a big deal.

Mhh, does it work thay way? Don't you need to bundle both because you don't know which will be imported? (But I agree this is not that important).

> and I think it's currently easier to keep it than refactoring it. Unfortunately, GCP has a limitation that the main cloud function file must be at the root

But how does this change things? You can keep the files in the same folders as of now, but use switch statement instead of dynamic import. Or am I missing something?

Reply:
Yeah encountered this GCP limitation when deploying with [serverless](https://www.serverless.com/framework/docs/providers/google/guide/functions)  

> Mhh, does it work thay way? Don't you need to bundle both because you don't know which will be imported? (But I agree this is not that important).

If the provider is specified in the config.json and also in the env variables then do we still need to be able to bundle both?

Reply:
> If the provider is specified in the config.json and also in the env variables then do we still need to be able to bundle both?

Yes, because we need to create bundles beforehand so they can be part of the package/Docker image. But during the deployment itself, we want to deploy just the specific one.